### PR TITLE
If anything happens during writing to the file, do not leave corrupt …

### PR DESCRIPTION
**Pull request by roesslerj**
…file

### CODE REVIEW

File: src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java [hunk 1]
```diff
@@ -68,13 +69,19 @@ private static Kryo createKryo() {
...
 		try ( final Output output = new Output( Files.newOutputStream( path ) ) ) {
 			output.writeString( version );
 			logger.debug( "Writing {} to {}. Do not write to same identifier or interrupt until done.", element,
 					identifier );
 			kryo.writeClassAndObject( output, element );
 			logger.debug( "Done writing {} to {}", element, identifier );
+		} catch ( Error | Exception anything ) {
+			logger.warn( "Error writing to file {}. Delete what has been writen to not leave corrupt file behind...",
+					identifier, anything );
+			FileUtil.deleteSecurely( file );
```
Comment (beatngu13) [inline src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java]: I suggest to remove `deleteSecurely` and reuse Apache's [`deleteQuietly`](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#deleteQuietly(java.io.File)), which we already use in `FilteUtil`:

>Deletes a file, never throwing an exception. If file is a directory, delete it and all sub-directories.

File: src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java [hunk 2]
```diff
@@ -68,13 +69,19 @@ private static Kryo createKryo() {
...
+		final File file = path.toFile();
 		FileUtil.ensureFolder( path.toFile() );
 		try ( final Output output = new Output( Files.newOutputStream( path ) ) ) {
 			output.writeString( version );
 			logger.debug( "Writing {} to {}. Do not write to same identifier or interrupt until done.", element,
 					identifier );
 			kryo.writeClassAndObject( output, element );
 			logger.debug( "Done writing {} to {}", element, identifier );
+		} catch ( Error | Exception anything ) {
+			logger.warn( "Error writing to file {}. Delete what has been writen to not leave corrupt file behind...",
```
Comment (beatngu13) [inline src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java]:
```suggestion
			logger.error( "Error writing to file '{}'. Deleting what has been written to not leave corrupt file behind...",
```

**Review by beatngu13 — CHANGES_REQUESTED** [review]

File: src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java (hunk 1 above)
Comment (roesslerj) [reply src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java]: Done

File: src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java (hunk 2 above)
Comment (roesslerj) [reply src/main/java/de/retest/recheck/persistence/bin/KryoPersistence.java]: Done

**Comment by beatngu13** [comment]
Test via the package-private ctor?

**Review by beatngu13 — APPROVED** [review]

**Pull request merged**